x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SslSettingsLoader.java

```diff
@@ -107,7 +107,7 @@ protected SslTrustConfig buildTrustConfig(Path basePath, SslVerificationMode ver
         if (trustRestrictions == null) {
             return trustConfig;
         }
-        throw new IllegalArgumentException("SSL trust_restrictions are not currently supported");
+        return new RestrictedTrustConfig(trustRestrictions, trustConfig);
```

Comment:
99% of this class was done & merged, but I couldn't add this line until `RestrictedTrustConfig` switched to implementing the new interface.